python-lib/ontology_tagging/sentence_splitter.py

```diff
@@ -39,6 +39,19 @@ def __init__(
     ):
         store_attr()
 
+    @staticmethod
+    def _clean_linebreaks(text: AnyStr) -> AnyStr:
```

Comment:
Method is a bit complicated, is there a more elegant way ?

Reply:
slightly simpler here 731a3b1

Reply:
Why is the filter needed?
Could you simply do
`"\n".join(elt.strip() for elt in text.splitlines())` ?

Reply:
Ah got it, filter is to remove the "" element if you have multiple line breaks ("\n\n").
I would recommend adding a comment to explain this.

Still, no need for list comprehension, a generator expression will be more memory efficient
`"\n".join(line.strip() for line in filter(None, text.splitlines()))`

Reply:
Yes you got it ! Good idea for the generator, thanks, did it here 8276993

Reply:
Almost there, just need to add a comment about why `filter(None, ...)`

Reply:
Oups I've missed this suggestion sorry, here it is :)  cc4d72f